### PR TITLE
CreatePlugin: Change default docker image to grafana-enterprise

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
We got a question about why the `GF_ENTERPRISE_*` environment variables wasn't picked up properly after migrating to the `create-plugin` configuration files. The reason for this is that we are using the `grafana/grafana` docker image which doesn't contain the enerprise version of Grafana.

This PR will change so that all the docker related configuration that is scaffolded by `create-plugin` will use `grafana/grafana-enterprise` as the default docker image. It also adds a way to configure this without adding any changes to the `./config` folder which might cause issues while updating.

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/create-plugin@0.10.0-canary.197.f95c2c8.0
  # or 
  yarn add @grafana/create-plugin@0.10.0-canary.197.f95c2c8.0
  ```
</details>

### CODE REVIEW

File: packages/create-plugin/templates/common/.config/README.md [hunk 1]
```diff
@@ -139,3 +139,26 @@ We need to update the `scripts` in the `package.json` to use the extended Webpac
 -"dev": "webpack -w -c ./.config/webpack/webpack.config.ts --env development",
 +"dev": "webpack -w -c ./webpack.config.ts --env development",
 ```
+
+### Configure grafana image to use when running docker
```
Comment (jackw) [inline packages/create-plugin/templates/common/.config/README.md:143]: Nice! 👏 

Could we also copy this to the [advanced config](https://github.com/grafana/plugin-tools/blob/main/docusaurus/docs/advanced-configuration.md) for the documentation website please?

File: packages/create-plugin/templates/common/.config/README.md [hunk 2]
```diff
@@ -139,3 +139,26 @@ We need to update the `scripts` in the `package.json` to use the extended Webpac
...
+version: '3.7'
+
+services:
+  grafana:
+    container_name: 'myorg-basic-app'
+    build:
+      context: ./.config
+      args:
+        grafana_version: ${GRAFANA_VERSION:-9.1.2}
+        grafana_image: ${GRAFANA_IMAGE:-grafana}
```
Comment (jackw) [inline packages/create-plugin/templates/common/.config/README.md:159]: I'm wondering if we should set the default `grafana_image` in the [docker-compose.yaml](https://github.com/grafana/plugin-tools/blob/main/packages/create-plugin/templates/common/docker-compose.yaml#L8-L9) to `grafana` and then update this doc to mention switching to `grafana-enterprise` with the diff for using `grafana-enterprise`?

e.g.

```
args:
  grafana_version: ${GRAFANA_VERSION:-9.1.2}
-  grafana_image: ${GRAFANA_IMAGE:-grafana}
+  grafana_image: ${GRAFANA_IMAGE:-grafana-enterprise}
```
Originally I thought switching to enterprise would be an easy solution to this but now I'm thinking it's more internal to Grafana and might confuse the community?

**Review by jackw — APPROVED** [review]
LGTM! 🚀 

Left a couple of nits.

File: packages/create-plugin/templates/common/.config/README.md (hunk 2 above)
Comment (mckn) [reply packages/create-plugin/templates/common/.config/README.md:159]: Will leave it as it is according to our chat, ok?

File: packages/create-plugin/templates/common/.config/README.md (hunk 1 above)
Comment (mckn) [reply packages/create-plugin/templates/common/.config/README.md:143]: Done

**Comment by jackw** [comment]
:shipit:

**Pull request merged**